pkg/operator/handler.go

```diff
@@ -79,6 +80,82 @@ type Handler struct {
 	generateDeployment generate.Generator
 }
 
+func isDeploymentStatusAvailable(o runtime.Object) bool {
+	switch deploy := o.(type) {
+	case *appsapi.DeploymentConfig:
+		for _, condition := range deploy.Status.Conditions {
+			if condition.Type != appsapi.DeploymentAvailable {
+				continue
+			}
+			return condition.Status == coreapi.ConditionTrue
+		}
+	case *kappsapi.Deployment:
+		for _, condition := range deploy.Status.Conditions {
+			if condition.Type != kappsapi.DeploymentAvailable {
+				continue
+			}
+			return condition.Status == coreapi.ConditionTrue
+		}
+	}
+	return false
+}
+
+func isDeploymentStatusProgressing(o runtime.Object) bool {
+	switch deploy := o.(type) {
+	case *appsapi.DeploymentConfig:
+		for _, condition := range deploy.Status.Conditions {
+			if condition.Type != appsapi.DeploymentProgressing {
+				continue
+			}
+			return condition.Status == coreapi.ConditionTrue
+		}
+	case *kappsapi.Deployment:
+		for _, condition := range deploy.Status.Conditions {
+			if condition.Type != kappsapi.DeploymentProgressing {
+				continue
+			}
+			return condition.Status == coreapi.ConditionTrue
+		}
+	}
+	return false
+}
+
+func (h *Handler) syncDeploymentStatus(cr *regopapi.ImageRegistry, o runtime.Object, statusChanged *bool) {
```

Comment:
There wasn't an issue with the Failing condition when @bparees and I looked at it yesterday, but I'm mildly curious why you left the updating of `osapi.OperatorFailing` in the main Handle method, but handle the other 2 here.

You could centralize things by adding err error as a parameter here and use this method where you are manipulating `osapi.OperatorFailing`.

Reply:
After discussing on the scram last week, we agreed that `ClusterOperator.Status.Available` and `ClusterOperator.Status.Processing` should reflect deployment status. `ClusterOperator.Status.Available=True` means that the `image-registry` is available for requests.

`OperatorFailing` is operator state. It is set depending on whether we update the resources.